src/psyclone/domain/gocean/transformations/gocean_opencl_trans.py

```diff
@@ -1018,7 +1020,7 @@ def _insert_ocl_arg_setter_routine(node, kernel):
         # Add the subroutine as child of the provided node
         node.addchild(argsetter)
 
-        return node.symbol_table.lookup_with_tag(sub_name)
+        return node.symbol_table.lookup_with_tag(kernel.name + "_set_args")
```

Comment:
I don't agree, the tag needs to be computable at entry to this function so should be `kernel.name + "_set_args" and not the result from next_available_name (since this is not consistent across multiple calls).

Reply:
I agree, it was my mistake, resolving issue